src/main/groovy/com/github/lookout/verspaetung/Main.groovy

```diff
@@ -53,74 +56,110 @@ class Main {
 
-        KafkaPoller poller = setupKafkaPoller(consumers, statsd, cli.hasOption('n'))
+        /* topicOffsets is where the KafkaPoller should be writing all of it's
+         * latest offsets from querying the Kafka brokers
+         */
+        ConcurrentHashMap<TopicPartition, Long> topicOffsets = new ConcurrentHashMap<>()
+
+        /* Hash map for keeping track of KafkaConsumer to ConsumerGauge
+         * instances. We're only really doing this because the MetricRegistry
+         * doesn't do a terrific job of exposing this for us
+         */
+        ConcurrentHashMap<KafkaConsumer, ConsumerGauge> consumerGauges = new ConcurrentHashMap<>()
+
+
+        KafkaPoller poller = new KafkaPoller(topicOffsets, watchedTopics)
         BrokerTreeWatcher brokerWatcher = new BrokerTreeWatcher(client).start()
-        StandardTreeWatcher consumerWatcher = new StandardTreeWatcher(client, consumers).start()
+        brokerWatcher.onBrokerUpdates << { brokers -> poller.refresh(brokers) }
+
+        poller.start()
+
+        /* Need to reuse this closure for the KafkaSpoutTreeWatcher if we have
+         * one
+         */
+        Closure gaugeRegistrar = { KafkaConsumer consumer ->
+            registerMetricFor(consumer, consumerGauges, consumerOffsets, topicOffsets)
+        }
+
+        StandardTreeWatcher consumerWatcher = new StandardTreeWatcher(client,
+                                                                      watchedTopics,
+                                                                      consumerOffsets)
+        consumerWatcher.onConsumerData << gaugeRegistrar
+        consumerWatcher.start()
+
 
         /* Assuming that most people aren't needing to run Storm-based watchers
          * as well
          */
         if (cli.hasOption('s')) {
-            KafkaSpoutTreeWatcher stormWatcher = new KafkaSpoutTreeWatcher(client, consumers)
+            KafkaSpoutTreeWatcher stormWatcher = new KafkaSpoutTreeWatcher(client,
+                                                                           watchedTopics,
+                                                                           consumerOffsets)
+            stormWatcher.onConsumerData << gaugeRegistrar
             stormWatcher.start()
         }
 
-        consumerWatcher.onInitComplete << {
-            logger.info("standard consumers initialized to ${consumers.size()} (topic, partition) tuples")
-        }
 
-        brokerWatcher.onBrokerUpdates << { brokers ->
-            poller.refresh(brokers)
+        if (cli.hasOption('n')) {
+            reporter = ConsoleReporter.forRegistry(registry)
+                                      .convertRatesTo(TimeUnit.SECONDS)
+                                      .convertDurationsTo(TimeUnit.MILLISECONDS)
+                                      .build()
         }
 
-        logger.info("Started wait loop...")
+        /* Start the reporter if we've got it */
+        reporter?.start(1, TimeUnit.SECONDS)
```

Comment:
This needs to be configurable in case it is reporting too frequently. Alternatively, report a little less often.